app/experimenter/kinto/tasks.py

```diff
@@ -78,54 +78,41 @@ def nimbus_check_kinto_push_queue():
     for application, collection in NimbusExperiment.KINTO_APPLICATION_COLLECTION.items():
         kinto_client = KintoClient(collection)
 
-        rejected_collection_data = kinto_client.get_rejected_collection_data()
-        if rejected_collection_data:
-            rejected_slug = kinto_client.get_rejected_record()
-            experiment = NimbusExperiment.objects.get(slug=rejected_slug)
-            if (
-                experiment.status == NimbusExperiment.Status.LIVE
-                and experiment.is_end_requested
-            ):
-                experiment.is_end_requested = False
-            else:
-                experiment.status = NimbusExperiment.Status.DRAFT
+        if kinto_client.has_pending_review():
+            continue
+        elif kinto_client.has_rejection():
+            handle_rejection(kinto_client)
+        elif queued_launch_experiment := NimbusExperiment.objects.launch_queue(
+            application
+        ).first():
+            nimbus_push_experiment_to_kinto.delay(queued_launch_experiment.id)
+        elif queued_end_experiment := NimbusExperiment.objects.end_queue(
```

Comment:
1) Move the pause check into the manager like above
2) Add the pause queue check here and dispatch a pause task (from below, pad it out the same way the end task is padded)
3) You can delete the update pause in kinto task altogether

Rename the push to, end in, pause in tasks to something similar like `handle_whatever`